cpp/src/dbscan/dbscan.hpp

```diff
@@ -19,6 +19,9 @@
 
 namespace ML {
 
+static const size_t DEFAULT_MAX_MEM_BYTES_D = 13e9;
+static const size_t DEFAULT_MAX_MEM_BYTES_S = 13e8;
```

Comment:
the single-precision max bytes looks a little low to me... why does it need to decrease so much? does it have a big impact on perf?
I'm in all in favor of supporting smaller gpus well, btw, if it doesn't impact features and perf noticeably...

Reply:
This was a temporary fix for a nasty overflow bug that occurs when using single-precision inputs. I was not sure if I'd have time to go through and refactor all the types this close to release, but our shiny new gtests made it much easier.